include/clang/AST/Expr.h

```diff
@@ -2798,7 +2798,9 @@ class BoundsExpr : public Expr {
 /// classes).
 class CastExpr : public Expr {
 private:
-  enum { OP, BOUNDS, END_EXPR = 2 };
+  // BOUNDS - enum value for bounds of cast expression
+  // SUBBOUNDS - enum value for bounds of subexpression
+  enum { OP, BOUNDS, SUBBOUNDS, END_EXPR = 3 };
```

Comment:
Okay, I will implement as follows:
SubExpr[SUBEXPRBOUNDS] - represent expanded/inferred bounds expression of subexpression
SubExpr[CASTBOUNDS] - represent expanded/inferred bounds expression of bounds cast operation
In code generation, it use those two kinds of bounds expression in bounds cast operation
Those are all inferred (expanded) from original bounds expression